core/src/main/java/org/infinispan/persistence/spi/NonBlockingStore.java

```diff
@@ -87,146 +83,145 @@
+    * Refer to {@link Characteristic} and its values for descriptions of each characteristic for stores.
     * @implSpec
-    * The default implementation returns an empty set
-    * @return the set of characteristics that this store supports
+    * The default implementation returns an empty set.
+    * @return the set of characteristics that this store supports.
     */
    default Set<Characteristic> characteristics() {
       return EnumSet.noneOf(Characteristic.class);
    }
 
    /**
-    * Returns a stage that when complete returns a boolean indicating whether the current store can be accessed for
+    * Returns a stage that, when complete, returns a boolean indicating whether the current store can be accessed for
     * requests. This can be useful for store implementations that rely on an external source, such as a remote database,
     * that may become unreachable. This can reduce sending requests to a store that is not available, as subsequent cache
     * requests will result in a {@link StoreUnavailableException} being thrown until the store becomes available again.
     * <p>
-    * Store availability is is polled periodically to update a store's status if it's availability changes. This method
-    * will not be invoked concurrently with itself (ie. this method will not be invoked until after the previous stage
-    * has completed), but will be invoked concurrently with other operations, excluding
+    * Store availability is is polled periodically to update the status of stores if their availability changes. This method
+    * is not invoked concurrently with itself. In other words, this method is not invoked until after the previous stage
+    * has completed. However this method is invoked concurrently with other operations, except for
     * {@link #start(InitializationContext)} and {@link #stop()}.
     * <p>
-    If a store is configured to be {@link StoreConfiguration#async()} and the store becomes unavailable, then it's
+    If a store is configured to be {@link StoreConfiguration#async()} and the store becomes unavailable, then it is
     possible for the cache operations to be accepted in the interim period between the loss of availability and the
     modification-queue becoming full. This allows for this store to be unavailable for short periods of time without a
-    {@link StoreUnavailableException} being thrown, however if the store does not become available before the queue
-    fills, then a {@link StoreUnavailableException} is eventually thrown.
+    {@link StoreUnavailableException} being thrown; however if the store does not become available before the queue
+    fills, then a {@link StoreUnavailableException} is thrown.
     * @implSpec
-    * The default implementations returns a completed stage with the value {@code Boolean.TRUE}
-    * @return stage that when complete signals if the store is available
+    * The default store implementations return a completed stage with the value {@code Boolean.TRUE}.
```

Comment:
@wburns could you also please check this line? "default store implementations return" or should it be "default store implementation returns"? I presume the former as Infinispan provides multiple store implementations.

Reply:
No this is referring to the default implementation of this method. TBH I think it should just be "The default implementation returns .."

Reply:
thanks @wburns I see that now :facepalm: was probably over-complicating things...